src/client.rs

```diff
@@ -176,23 +176,23 @@ fn from_process(process: &str) -> Option<Vec<String>> {
 }
 
 fn parse_process(value: &str) -> Result<(String, String)> {
-    let re = regex::Regex::new(r#"--remoting-auth-token="?([\w]*)"?.*--app-port="?([0-9]*)"?"#)
-        .or(Err(Error::AppNotRunning))?;
-    let caps = re.captures(value);
-    let caps = caps.ok_or(Error::AppNotRunning)?;
-
-    let token: String = caps
-        .get(1)
-        .ok_or(Error::AppNotRunning)?
-        .as_str()
-        .to_string();
-    let port: String = caps
-        .get(2)
-        .ok_or(Error::AppNotRunning)?
-        .as_str()
-        .to_string();
-
-    Ok((token, port))
+    let args: Vec<&str> = value.split("\"").collect();
+
+    let mut token = Err(Error::AppNotRunning);
+    let mut port = Err(Error::AppNotRunning);
+
+    for arg in args {
+        if arg.starts_with("--remoting-auth-token") {
+            let val = arg.split("=").collect::<Vec<&str>>()[1];
```

Comment:
mind getting rid of the index here? 

```rust
let val = arg.split("=").collect::<Vec<&str>>().first();
let token = val.ok_or(Error::AppNotRunning);
```

Reply:
Yeah I'll change it no problem, though if I understand how args work it should always split into exactly 2 parts so this should still be fine I think.

Reply:
you're probably right. This is just the only thing that's bitten us at work lately so i try to avoid it as much as possible.

Reply:
Now that I changed it, it looks cleaner without indexing, plus it avoids needing to make a vector. I'm kind of new to rust still so I overcomplicate stuff like this a lot.